data/yam/agama/auto/template.libsonnet

```diff
@@ -9,15 +9,17 @@ function(addon_ha_reg_code='',
          user=true,
          root=true,
          storage='',
+         packages='',
          patterns='',
          product='',
          registration_code='',
          scripts_pre='',
          scripts_post='') {
   [if bootloader == true then 'bootloader']: base_lib['bootloader'],
-  [if patterns != '' then 'software']: {
-    patterns: [patterns]
-  },
+  [if patterns != '' || packages != '' then 'software']: std.prune({
+    patterns: if patterns != '' then std.split(patterns, ','),
+    packages: if packages != '' then std.split(packages, ','),
```

Comment:
In case you set `--tla-str patterns=one,two` and no packages variable set. What is the output json? just to check if packages will be empty or not set in the resulting JSON

Reply:
> In case you set `--tla-str patterns=one,two` and no packages variable set. What is the output json? just to check if packages will be empty or not set in the resulting JSON

$jsonnet  --tla-str product=SLES  --tla-str registration_code=xxxx --tla-str scripts_post=enable_root_login  --tla-code bootloader=true  --tla-str patterns=a,b,c   data/yam/agama/auto/template.libsonnet
```